apps/emqx/src/emqx_datetime.erl

```diff
@@ -47,7 +52,7 @@ to_epoch_millisecond(DateTime) ->
 to_epoch(DateTime, Unit) ->
     try
         case string:to_integer(DateTime) of
-            {Epoch, []} when Epoch >= 0 -> {ok, Epoch};
+            {Epoch, []} when Epoch >= 0, Epoch =< ?MAXIMUM_EPOCH -> {ok, Epoch};
```

Comment:
have to take Unit into consideration ?

Reply:
ah sorry, it's datetime convert to epoch.

Reply:
the is a Unix timestamp, so the unit is `second` and it also is the default unit for `calendar:system_time_to_rfc3999/1`, even if misused with other units when calling `calendar:system_time_to_rfc3999/1`,  the value only becomes smaller and safer

Reply:
Sorry again.  Help me understand this.

If this is a string format of epoch on 9999-12-31, it would be `"253402214400"` which is valid.
If it's at millisecond level, it would be `"253402214400000"` which is invalid ?

Reply:
It seems `epoch_millisecond()` type is used in quite a few hocon schema fields.

Reply:
so there should be a `-define(MAXIMUM_EPOCH_MILLI, 253402214400_000).` and check the unit here.

Reply:
Yes, you are right, the unit is important because both the second and millisecond call this function